torch/distributed/nn/api/remote_module.py

```diff
@@ -35,6 +46,15 @@ def _create_module(module_cls, args, kwargs, module_interface_cls=None):
         )
     if module_interface_cls is not None:
         module = torch.jit.script(module)
+    if device:
+        d = torch.device(device)
+        if d.type == "cuda" and d.index >= torch.cuda.device_count():
+            raise RuntimeError(
```

Comment:
`torch.device(` would already cover any errors and throw an exception, why do we need this additional line? Also, is it sufficient to just call `.to()` here and would it throw the appropriate exception?  Or does `.to()` not throw an appropriate exception.

Reply:
For the input string like "cuda:100", `.to()` will not throw an exception until some real computation is deployed. Shall we move such check to module.to()?

Reply:
Please ignore the above comment. You are totally correct.

This is handled by correctly by PyTorch lib, confirmed by devgpu. Previously I didn't have the access to devgpu, so I used a Colab to test the error message, which may not have the most recent version installed.